cylc/uiserver/resolvers.py

```diff
@@ -15,10 +15,189 @@
 
 """GraphQL resolvers for use in data accessing and mutation of workflows."""
 
+import getpass
+import logging
+import os
+from shlex import quote
+from subprocess import Popen, PIPE, DEVNULL
+
 from cylc.flow.network.resolvers import BaseResolvers
 from cylc.flow.data_store_mgr import WORKFLOW
 
 
+DEBUG = True
+logger = logging.getLogger(__name__)
+
+
+def snake_to_kebab(snake):
+    """Convert snake_case text to --kebab-case text.
+
+    Examples:
+        >>> snake_to_kebab('foo_bar_baz')
+        '--foo-bar-baz'
+        >>> snake_to_kebab('')
+        ''
+        >>> snake_to_kebab(None)
+        Traceback (most recent call last):
+        TypeError: <class 'NoneType'>
+
+    """
+    if isinstance(snake, str):
+        if not snake:
+            return ''
+        return f'--{snake.replace("_", "-")}'
+    raise TypeError(type(snake))
+
+
+def check_cylc_version(version):
+    """Check the provided Cylc version is available on the CLI.
+
+    Sets CYLC_VERSION=version and tests the result of cylc --version
+    to make sure the requested version is installed and selectable via
+    the CYLC_VERSION environment variable.
+    """
+    proc = Popen(
+        ['cylc', '--version', '--color=never'],
+        env={**os.environ, 'CYLC_VERSION': version},
+        stdin=DEVNULL,
+        stdout=PIPE,
+        stderr=PIPE,
+        text=True
+    )
+    ret = proc.wait(timeout=5)
+    out, err = proc.communicate()
+    return ret or out.strip() == version
```

Comment:
I tested the function above in the IDE (checked out the branch, "Python Console" auto-imports the venv and I can play with the new code) and it always returned `1` (not a bool, but no biggie I think).

![image](https://user-images.githubusercontent.com/304786/110031399-bc732e00-7d9b-11eb-842d-e28f37f1c8ac.png)

I tried the command above and it failed too.

```py
(venv) kinow@ranma:~/Development/python/workspace/cylc-uiserver$ cylc --version --color=never
Error: cylc --color=never: unknown utility. Abort.
Type "cylc help all" for a list of utilities.
```

I think we need to replace `--version` with `version`, i.e. `cylc version --color=never`?

![Screenshot from 2021-03-05 10-17-50](https://user-images.githubusercontent.com/304786/110031667-11af3f80-7d9c-11eb-8af2-9b26a137e852.png)

Reply:
Good testing!

I added the `--color=never` for safety as an afterthought, clearly not as safe as I had thought!